### ISSUE TITLE
Add a deprecation warning in the Bonsai package manager

### ISSUE THREAD

**Issue by cjsha** [bonsai-rx/bonsai#1965]
You can see a package that is marked as deprecated in nuget, but the only way I know that it is deprecated through the Bonsai UI is if I read its description which doesn't really pop out at me. And that deprecation warning is manually added - not all deprecated packages will have descriptions that indicate as such.

![bonsai-deprecated-package](https://github.com/user-attachments/assets/37f6c891-2675-480c-8e37-1da68093d303)

**Comment by PathogenDavid** [comment]
For reference, here's how the Visual Studio NuGet package manager shows deprecation:

![image](https://github.com/user-attachments/assets/23a56e04-6e95-4e62-863b-84a15ae32f3a)

**Comment by glopesdev** [comment]
@cjsha It looks like it is possible to get the custom deprecation message via `GetDeprecationMetadataAsync`. Below is a proposed layout to resolve this.

I tried to strike a balanced tone with the current package details dialog. What I mostly don't like is there is no good place to insert the warning icon in the left package view. Ideally we would migrate the UI to more closely mimic the modern package view, with the listed version numbers, and use only icons instead of buttons, but it's a significant amount of work to make an unrelated feature look slightly better.

There are also two more properties in `PackageDeprecationMetadata`: a list of `Reasons` and an `AlternatePackage` which are probably used to fill the "More info" section in the dialog. I think we would not include these on a first release of the deprecation UI, or alternately we can try to carve out space to render a standard deprecation message listing the alternate package at the bottom of the description.

![image](https://github.com/user-attachments/assets/ba1daca7-c826-4d9f-b628-4d70df888319)

**Comment by cjsha** [comment]
> I tried to strike a balanced tone with the current package details dialog. What I mostly don't like is there is no good place to insert the warning icon in the left package view. Ideally we would migrate the UI to more closely mimic the modern package view, with the listed version numbers, and use only icons instead of buttons, but it's a significant amount of work to make an unrelated feature look slightly better.

It would be nice for someone to know it's deprecated before clicking on it because I expect most people to click and install without really looking much at what's on the right side, but I'm also sympathetic to aesthetic sensibilities. It seems like this point is outdated though if I'm to understand what you describe in [this PR](https://github.com/bonsai-rx/bonsai/pull/1969). 

If there's going to be an ``AlternatePackage`` section, I would recommend carving out space for it underneath the deprecation notice if possible. To me, those two items belong together. 

Could the ``AlternatePackage`` section include a button for automatically searching/selecting the alternate package for installation? That seems like a lot more work, but I figure I'd throw it out for your consideration. Although it's more ideal to use the dedicated ``AlternatePackage`` tag for conveying what alternative package the user can use, I suppose it's sufficient to do that for now in the deprecation message that occurs under the "Deprecation" header like you did in that screenshot. For that reason, I suppose it's ok to leave the ``AlternatePackage`` section out for now if that's still part of your plan.

**Comment by glopesdev** [comment]
> Could the AlternatePackage section include a button for automatically searching/selecting the alternate package for installation?

Yup, this would be easy if we  are consistent with the modern VS package manager, where clicking on the alternate package makes a search by that package ID.

Now that I have bitten the bullet and revamped most of the details page it probably won't be too hard to add the alternate package section.
